src/ngraph/op/fused/group_conv.cpp

```diff
@@ -225,77 +253,145 @@ void op::v1::GroupConvolutionBackpropData::set_output_shape(const Shape& shape)
-                               m_strides,
-                               m_dilations,
-                               m_auto_pad,
-                               m_pads_end,
-                               m_pads_begin);
+            m_pads_begin = conv_default_padding(this, data_pshape, filters_pshape);
+        }
+        if (m_pads_end.size() == 0)
+        {
+            m_pads_end = conv_default_padding(this, data_pshape, filters_pshape);
+        }
+        if (m_output_padding.size() == 0)
+        {
+            m_output_padding = conv_default_padding(this, data_pshape, filters_pshape);
+        }
+        if (m_strides.size() == 0)
+        {
+            m_strides = conv_default_strides(this, data_pshape, filters_pshape);
         }
+        if (m_dilations.size() == 0)
+        {
+            m_dilations = conv_default_strides(this, data_pshape, filters_pshape);
+        }
+
+        const size_t num_spatial_dims = data_shape.size() - 2;
+
+        NODE_VALIDATION_CHECK(this,
+                              m_strides.size() == num_spatial_dims,
+                              "Strides should be defined for all and only spatial features.");
+
+        NODE_VALIDATION_CHECK(this,
+                              m_dilations.size() == num_spatial_dims,
+                              "Dilations should be defined for all and only spatial features.");
+
+        NODE_VALIDATION_CHECK(this,
+                              m_output_padding.size() == num_spatial_dims,
+                              "Output padding should be defined for all and only "
+                              "spatial features.");
     }
 
-    PartialShape result_shape;
+    bool is_output_shape_present = get_inputs().size() == 3;
+    PartialShape output_pshape;
+
+    // If output shape is provided, ignore current values for padding begin/end
+    // and infer them.
     if (is_output_shape_present)
     {
+        output_pshape = get_output_shape();
+
+        // If auto_pad has one of following mode we infer paddings. Otherwise in EXPLICIT auto_pad
+        // mode we use what is provided.
+        if ((m_auto_pad == PadType::SAME_UPPER || m_auto_pad == PadType::SAME_LOWER) &&
+            output_pshape.is_static() && data_pshape.is_static() && filters_pshape.is_static())
+        {
+            Shape output_shape = output_pshape.to_shape();
+            // extract spatial shape values
+            // output_shape.erase(std::begin(output_shape), std::next(std::begin(output_shape), 2));
+            const Shape& data_shape = data_pshape.to_shape();
+            const Shape& filters_shape = filters_pshape.to_shape();
+            const size_t num_spatial_dims = data_shape.size() - 2;
+            NODE_VALIDATION_CHECK(this,
+                                  output_shape.size() - 2 == num_spatial_dims,
+                                  "Output shape should be specified only and for "
+                                  "all spatial dimensions.");
```

Comment:
I think that both this check and the statement you mentioned are working since the `get_output_shape` inserts values for batch and channel dimensions. However I came into conclusion that the method `get_output_shape()` should return dimensions only for spatial features as it is defined in the documentation.

Reply:
Ou i missed this little `- 2` sorry for that(maybe I get confused by validation message).
Since you have mentioned that ONNX specify only spatial dimensions for `output_shape` we can update this method.